client/my-sites/earn/ads/index.jsx

```diff
@@ -0,0 +1,392 @@
+	}
+
+	renderEmptyContent() {
+		return (
+			<EmptyContent
+				illustration="/calypso/images/illustrations/illustration-404.svg"
+				title={ this.props.translate( 'You are not authorized to view this page' ) }
+			/>
+		);
+	}
+
+	render() {
+		const { site, translate, isUnsafe, wordAdsError } = this.props;
+		const contentUnsafe = [ 'mature', 'spam', 'other' ].includes( this.isUnsafe );
+		const jetpackPremium =
+			site.jetpack &&
+			( isPremium( site.plan ) || isBusiness( site.plan ) || isEcommerce( site.plan ) );
+
+		let notice = null;
+		let component = this.props.children;
+
+		// No access.
+		if ( ! canAccessAds( site ) ) {
+			return <div>{ this.renderEmptyContent() }</div>;
+		}
+
+		// Error for non-admins.
+		if ( ! userCan( 'manage_options', this.props.site ) ) {
+			return (
+				<div>
+					<Notice
+						status="is-warning"
+						text={ this.props.translate( 'Only site administrators can edit Ads settings.' ) }
+						showDismiss={ false }
+					/>
+				</div>
+			);
+		}
+
+		// Has not applied.
+		// @TODO do jetpack sites have this option set?
+		if ( ! site.options.wordads ) {
+			// If site has qualifying plan.
+			if ( isWordadsInstantActivationEligible( site ) ) {
+				// Application.
+				return <div>Application goes here.</div>;
+			} else {
+				// @TODO do we need this? if ( canUpgradeToUseWordAds( site ) ) {
+
+				return <div>{ this.renderUpsell() }</div>;
+			}
+		} else {
+			// If Error.
+			if ( this.wordAdsError ) {
+				return (
+					<Notice
+						classname="ads__activate-notice"
+						status="is-error"
+						onDismissClick={ this.handleDismissWordAdsError }
+					>
+						{ this.props.wordAdsError }
+					</Notice>
+				);
+			}
+
+			// Not safe.
+			if ( this.contentUnsafe ) {
+				return <div>{ this.renderBannerApplicationDenyBrandSafety() }</div>;
+			}
+
+			// If private.
+			if ( this.props.isUnsafe === 'private' ) {
+				return (
+					<Notice
+						classname="ads__activate-notice"
+						status="is-warning"
+						showDismiss={ false }
+						text={ translate(
+							'Your site is marked as private. It needs to be public so that visitors can see the ads.'
+						) }
+					>
+						<NoticeAction href={ '/settings/general/' + this.props.siteSlug }>
+							{ translate( 'Change privacy settings' ) }
+						</NoticeAction>
+					</Notice>
+				);
+			}
+
+			/*
+			if ( 'Application is pending / in review' ) {
+				return (
+					<div>
+						{ this.renderBannerApplicationReview() }
+						{
+							// @TODO add new application
+						}
+					</div>
+				);
+			}
+
+			elseif ( 'application is rejected for traffic too low' ) {
+				return (
+					<div>
+						{ this.renderBannerApplicationDenyTraffic() }
+						{
+							// @TODO add new application
+						}
+					</div>
+				);
+			}
+			*/
+			return (
+				<div>
+					{ // Recently approved.
+					this.props.wordAdsSuccess && this.renderBannerApplicationApprove() }
+
+					{
+						// Regular display for WordAds Users.
+						component
```

Comment:
How is this displaying the form from form-settings.jsx?

Reply:
On line 263 `component` is set to `this.props.children`. `this.props.children` are any elements that are being wrapped by the current element. In this case it's `<AdsSettings />`:

From `earn/main.jsx` line 74
```
<AdsSection section={ this.props.section }>
    <AdsSettings /> //this.props.children
</AdsSection>
```